lib/queues.js

```diff
@@ -260,16 +270,64 @@ class Queue {
   runOnce() {
     assert(!this._processing, 'Cannot call once while continuously processing jobs');
     const hasHandlers = (this._handlers.size > 0);
+
     if (hasHandlers) {
 
       debug('Waiting for jobs on queue %s', this.name);
-      return this._getClientPromise.then(client => this._reserveAndProcess(client));
+
+      if (this._delayedJobs.size > 0) {
+
+        const jobIDs = this._getDelayedJobIDs();
+
+        return this._kickJobs(jobIDs)
+          .then(() => this._getClientPromise)
+          .then(client => this._reserveAndProcess(client));
+
+      } else {
+
+        return this._getClientPromise
+          .then(client => this._reserveAndProcess(client));
+
+      }
 
     } else
       return Promise.resolve(false);
   }
 
 
+  // Returns a list of stored delayed jobs IDs that are ready to be processed
+  _getDelayedJobIDs() {
```

Comment:
So per the comment above the function probably needs the "ready" term in its name?